src/js/game/LevelMVC/LevelView.js

```diff
@@ -2096,28 +2099,69 @@ module.exports = class LevelView {
-        frameList = Phaser.Animation.generateFrameNames("redstone_sparkle", 0, 24, ".png");
-        for (i = 0; i < 4; ++i) {
-          frameList.push("redstone_sparkle7");
+        let posX = xOffset + 40 * x;
+        let posY = yOffset + group.yOffset + 40 * y;
+        sprite = group.create(posX, posY, atlas, frame);
+        // Establish the three different animations.
+        let anim = [];
+        let whichAnim = Math.floor(Math.random() * 3);
+        // Pick one of the animations to assign.
+        switch (whichAnim) {
+          case 0:
+            anim = Phaser.Animation.generateFrameNames("redstone_sparkle", 0, 7, ".png");
+            break;
+          case 1:
+            anim = Phaser.Animation.generateFrameNames("redstone_sparkle", 8, 15, ".png");
+            break;
+          default:
+            anim = Phaser.Animation.generateFrameNames("redstone_sparkle", 16, 23, ".png");
+            break;
         }
-        frameList = frameList.concat(Phaser.Animation.generateFrameNames("redstone_sparkle", 8, 13, ".png"));
-        for (i = 0; i < 3; ++i) {
-          frameList.push("redstone_sparkle13");
+
+        // Set up a random amount of blank frames to be put on the front and back for delay between loops.
+        let frontDelay = Math.floor(Math.random() * 50) + 5;
+        let backDelay = Math.floor(Math.random() * 30) + 20;
+        let frontBuffer = [];
+        let backBuffer = [];
+        for (i = 0; i < frontDelay; ++i) {
+          frontBuffer.push("redstone_sparkle99.png");
         }
-        frameList = frameList.concat(Phaser.Animation.generateFrameNames("redstone_sparkle", 14, 23, ".png"));
-        for (i = 0; i < 8; ++i) {
-          frameList.push("redstone_sparkle1");
+        for (i = 0; i < backDelay; ++i) {
+          backBuffer.push("redstone_sparkle99.png");
         }
-        sprite.animations.add("idle", frameList, 5, true);
+
+        // Organize the animation with a buffer in front and back
+        frameList = frontBuffer.concat(frameList);
+        frameList = frameList.concat(anim);
+        frameList = frameList.concat(backBuffer);
+
+        let animation = sprite.animations.add("idle", frameList, 5, true);
+
+        // onLoop, we want to randomize which corner of the index the animation manifests in.
+        animation.onLoop.add((sprite) => {
+          let xType = Math.floor(Math.random() * 2);
+          let yType = Math.floor(Math.random() * 2);
+          if (xType) {
+            sprite.position.x = sprite.cameraOffset.x + 20;
+          } else {
+            sprite.position.x = sprite.cameraOffset.x;
+          }
+          if (yType) {
+            sprite.position.y = sprite.cameraOffset.y + 20;
+          } else {
+            sprite.position.y = sprite.cameraOffset.y;
+          }
+        }, this);
+
+        // Play that thing!
         this.playAnimationWithOffset(sprite, "idle", Math.floor(Math.random() * 3) + 21, 1);
         break;
-
+      }
```

Comment:
Nit: braces around case statement aren't needed.

Reply:
I didn't think so either, but the linter won't let that fly, given that I declare local scoped variables like:
        let frontBuffer = [];
        let backBuffer = [];

you get the error:
2103:7   error  Unexpected lexical declaration in case block  no-case-declarations

Reply:
Ah, interesting.  That makes sense then!